kornia/geometry/ransac.py

```diff
@@ -0,0 +1,224 @@
+               batch_size: int, device=torch.device('cpu')) -> torch.Tensor:
+        """Minimal sampler, but unlike traditional RANSAC we sample in batches to get benefit of the parallel
+        processing, esp.
+
+        on GPU
+        """
+        out: torch.Tensor = torch.empty(batch_size, sample_size)
+        # for loop, until https://github.com/pytorch/pytorch/issues/42502 accepted
+        for i in range(batch_size):
+            out[i] = torch.randperm(pop_size, dtype=torch.int32, device=device)[:sample_size]
+        return out
+
+    @staticmethod
+    def max_samples_by_conf(n_inl: int, num_tc: int, sample_size: int, conf: float):
+        """Formula to update max_iter in order to stop iterations earlier
+        https://en.wikipedia.org/wiki/Random_sample_consensus."""
+        if n_inl == num_tc:
+            return 1
+        return math.log(1.0 - conf) / math.log(1. - math.pow(n_inl / num_tc, sample_size))
+
+    def estimate_model_from_minsample(self,
+                                      kp1: torch.Tensor,
+                                      kp2: torch.Tensor) -> torch.Tensor:
+        batch_size, sample_size = kp1.shape[:2]
+        H = self.minimal_solver(kp1,
+                                kp2,
+                                torch.ones(batch_size,
+                                           sample_size,
+                                           dtype=kp1.dtype,
+                                           device=kp1.device))
+        return H
+
+    def verify(self,
+               kp1: torch.Tensor,
+               kp2: torch.Tensor,
+               models: torch.Tensor, inl_th: float) -> Tuple[torch.Tensor, torch.Tensor, float]:
+        if len(kp1.shape) == 2:
+            kp1 = kp1[None]
+        if len(kp2.shape) == 2:
+            kp2 = kp2[None]
+        batch_size = models.shape[0]
+        errors = self.error_fn(kp1.expand(batch_size, -1, 2),
+                               kp2.expand(batch_size, -1, 2),
+                               models)
+        inl = (errors <= inl_th)
+        models_score = inl.float().sum(dim=1)
+        best_model_idx = models_score.argmax()
+        best_model_score = models_score[best_model_idx].item()
+        model_best = models[best_model_idx].clone()
+        inliers_best = inl[best_model_idx]
+        return model_best, inliers_best, best_model_score
+
+    def remove_bad_samples(self, kp1: torch.Tensor, kp2: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        """"""
+        # ToDo: add (model-specific) verification of the samples,
+        # E.g. contraints on not to be a degenerate sample
+        return kp1, kp2
+
+    def remove_bad_models(self, models: torch.Tensor) -> torch.Tensor:
+        # ToDo: add more and better degenerate model rejection
+        # For now it is simple and hardcoded
+        main_diagonal = torch.diagonal(models,
+                                       dim1=1,
+                                       dim2=2)
```

Comment:
```suggestion
                                       dim2=2, device=self.device)
```
maybe we should store device